sources/packages/backend/apps/workers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -49,15 +53,41 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
           ASSESSMENT_NOT_FOUND,
         );
       }
+      // Check if the assessment is in one of the initial statuses. Any status different than "Submitted" and "Queued" should
+      // cancel the workflow because there is no point executing it.
+      // A workflow in these statuses can be associated and updated to "In progress" nicely. The expectation is that it would
+      // be primarily "Queued" but there is no harm in allowing "Submitted" also.
```

Comment:
Considering submitted here is to keep the option of starting the assessments manually for any troubleshooting purpose if required right?

Reply:
Yes, you are right. The concern here was more about not allowing the association to other statuses.